Build issue on `HAVE_UNUSED_ATTRIBUTE` when cross-compiling

Hello,

When trying to cross compile this lib with `cmake` I face this error:

```
-- Performing Test HAVE_UNUSED_ATTRIBUTE
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_UNUSED_ATTRIBUTE_EXITCODE (advanced)
   HAVE_UNUSED_ATTRIBUTE_EXITCODE__TRYRUN_OUTPUT (advanced)
For details see [...]libdwarf-20210528_build/TryRunResults.cmake
-- Performing Test HAVE_UNUSED_ATTRIBUTE - Failed
-- Checking compiler supports __attribute__ unused... PLEASE_FILL_OUT-FAILED_TO_RUN
```

The code is here : https://github.com/davea42/libdwarf-code/blob/master/CMakeLists.txt#L228

I think the fix could be to change the check from `check_c_source_runs` to `check_c_source_compiles`, as the `unused` attribute is a compilation check.

What do you think ?

Thank you

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -225,8 +225,8 @@ check_c_source_compiles("
   }" HAVE_RAW_LIBELF_OK)
 else()
 endif()
-check_c_source_runs("
   static unsigned foo( unsigned x, 
+check_c_source_compiles("
       __attribute__ ((unused)) int y)
   {  
       unsigned x2 = x + 1;

